src/commands/kv/bucket/mod.rs

```diff
@@ -128,3 +160,87 @@ fn get_digest(value: String) -> Result<String, failure::Error> {
     let hex_digest = HEXLOWER.encode(digest.as_ref());
     Ok(hex_digest)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::fs;
+
+    use walkdir::WalkDir;
+
+    #[test]
+    fn it_can_ignore_dir() {
+        let dir_name = "node_modules";
+        // If test dir already exists, delete it.
+        if fs::metadata(dir_name).is_ok() {
+            fs::remove_dir_all(dir_name).unwrap();
+        }
+
+        fs::create_dir(dir_name).unwrap();
+        fs::File::create(format!("{}/ignore_me.txt", dir_name)).unwrap();
+
+        let mut actual_count = 0;
+        for _ in WalkDir::new(dir_name)
+            .into_iter()
+            .filter_entry(|e| !is_ignored(e))
+        {
+            actual_count = actual_count + 1;
+        }
+
+        fs::remove_dir_all(dir_name).unwrap();
+
+        // No iterations should happen above because "node_modules" and its contents are ignored.
+        let expected_count = 0;
+        assert!(actual_count == expected_count);
+    }
+
+    #[test]
+    fn it_can_ignore_prefix() {
+        let file_name = ".dotfile";
+        // If test file already exists, delete it.
+        if fs::metadata(file_name).is_ok() {
+            fs::remove_file(file_name).unwrap();
+        }
+
+        fs::File::create(file_name).unwrap();
```

Comment:
do these tests create a real file in the file system or just something in memory? would be great if we didn't add more tests relying on file system but i'm not going to block on this

Reply:
I tried using tempfiles but had absolutely no success :( It seems that the tempfiles crate is NOT for testing purposes like mine (it produces temp gobbledygook directory names instead of allowing you to create a directory with the name you want, e.g. `node_modules`). So, I had to revert to using a real file in the filesystem.

Reply:
ok :)

Reply:
agree, but also can say this is a major RM planned for Q4